src/core/tests/components/test_input_populate.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+from ..schema_validator import get_validator_for
+from workflows_engine.exceptions import InvalidArguments
+from workflows_engine.validators.basic import greater_than_zero
+from workflows_engine.core.containers import Populate
+
+
+def test_populate_path():
+    """ Test populate class with optionals path parameters """
+    populate = Populate(
+        validators=[greater_than_zero()],
+        path="$.pick_details.quants[?(@.product.barcode == 'product')].qty",
+    )
+    validator = get_validator_for("components/populate")
+    validator.validate(populate.as_dict())
+
+
+def test_populate_value():
+    """ Test populate class with optional value parameters """
+    populate = Populate(validators=[greater_than_zero()], value=12)
+    validator = get_validator_for("components/populate")
+    validator.validate(populate.as_dict())
+
+
+def test_populate_with_exception():
+    """ Test populate exceptions raised by wrong parameters """
+    # Populate class with all parameters
+    with pytest.raises(InvalidArguments):
+        Populate(
+            validators=[greater_than_zero()],
+            path="$.pick_details.quants[?(@.product.barcode == 'product')].qty",
+            value=12,
+        )
+    # Populate class without path and value parameters
+    with pytest.raises(InvalidArguments):
```

Comment:
Split into two tests otherwise, it would take at least two runs to know they're both broken.

Reply:
done